### PR TITLE
Adjust auction price to `32 * Momp`

### PR DESCRIPTION
**Pull request by ith-harvey**
- Adjusted auction price
- Altered tests to reflect change

### CODE REVIEW

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 1]
```diff
@@ -101,7 +101,7 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
         _borrow(
             {
                 from:       _borrower2,
-                amount:     7_980 * 1e18,
+                amount:     7_980.0 * 1e18,
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: the .0 can be removed

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 2]
```diff
@@ -172,7 +172,7 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
                 bondSize:       0,
                 bondFactor:     0,
                 kickTime:       0,
-                kickPriceIndex: 0
+                kickMomp:       0
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: NIT: `_assertAuction` alignment, remove extra spaces that were added for `kickPriceIndex`

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 3]
```diff
@@ -241,7 +241,7 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
                 bondSize:       0.195342779771472726 * 1e18,
                 bondFactor:     0.01 * 1e18,
                 kickTime:       block.timestamp,
-                kickPriceIndex: 3696
+                kickMomp:       9.721295865031779605 * 1e18
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: NIT: same as above, `_assertAuction` alignment

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 4]
```diff
@@ -280,7 +280,7 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
                 bondSize:       0,
                 bondFactor:     0,
                 kickTime:       0,
-                kickPriceIndex: 0
+                kickMomp: 0
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: same as above, `_assertAuction` alignment

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 5]
```diff
@@ -304,7 +304,7 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
                 bondSize:       0.195342779771472726 * 1e18,
                 bondFactor:     0.01 * 1e18,
                 kickTime:       block.timestamp,
-                kickPriceIndex: 3696
+                kickMomp:       9.721295865031779605 * 1e18
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: same as above,` _assertAuction` alignment

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 6]
```diff
@@ -332,7 +332,7 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
                 bondSize:       0,
                 bondFactor:     0,
                 kickTime:       0,
-                kickPriceIndex: 0
+                kickMomp: 0
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: same as above, `_assertAuction` alignment

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 7]
```diff
@@ -354,7 +354,7 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
...
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: same as above, `_assertAuction` alignment

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 8]
```diff
@@ -378,7 +378,7 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
                 bondSize:       0.195342779771472726 * 1e18,
                 bondFactor:     0.01 * 1e18,
                 kickTime:       block.timestamp,
-                kickPriceIndex: 3696
+                kickMomp:       9.721295865031779605 * 1e18
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: same as above, `_assertAuction` alignment

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 9]
```diff
@@ -404,7 +404,7 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
                 bondSize:       0,
                 bondFactor:     0,
                 kickTime:       0,
-                kickPriceIndex: 0
+                kickMomp: 0
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: same as above, `_assertAuction` alignment

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 10]
```diff
@@ -426,7 +426,7 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
...
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: same as above, `_assertAuction` alignment

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 11]
```diff
@@ -450,7 +450,7 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
                 bondSize:       0.195342779771472726 * 1e18,
                 bondFactor:     0.01 * 1e18,
                 kickTime:       block.timestamp,
-                kickPriceIndex: 3696
+                kickMomp:       9.721295865031779605 * 1e18
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: same as above, `_assertAuction` alignment

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 12]
```diff
@@ -494,7 +494,7 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
...
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: `_assertAuction` alignment

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 13]
```diff
@@ -526,13 +526,26 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
                 bondSize:       0,
                 bondFactor:     0,
                 kickTime:       0,
-                kickPriceIndex: 0
+                kickMomp:       0
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: `_assertAuction` alignment

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 14]
```diff
@@ -526,13 +526,26 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
...
+                borrowerCollateralization: 1.0 * 1e18,
+                borrowerPendingDebt:       0 * 1e18
+            }
+        );
+
         _assertKicker(
             {
                 kicker:    _lender,
-                claimable: 0, // the entire bond was penalized
+                claimable: 0.0 * 1e18, // the entire bond was penalized
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: `0.0 * 1e18` can be made `0`

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 15]
```diff
@@ -580,14 +593,14 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
                 bondSize:       98.533942419792216457 * 1e18,
                 bondFactor:     0.01 * 1e18,
                 kickTime:       block.timestamp,
-                kickPriceIndex: 3696
+                kickMomp:       9.721295865031779605 * 1e18
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: alignment

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 16]
```diff
@@ -580,14 +593,14 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
...
             }
         );
         _assertKicker(
             {
                 kicker:    _lender,
                 claimable: 0,
-                locked:    98.533942419792216457 * 1e18
+                locked:   98.533942419792216457 * 1e18
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: alignment, need an extra space

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 17]
```diff
@@ -603,17 +616,17 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
         );
 
         //skip ahead so take can be called on the loan
-        skip(5 hours);
+        skip(10 hours);
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol:619]: why was this changed from 5 to 10 hours?

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 18]
```diff
@@ -603,17 +616,17 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
...
 
-        // perform partial take for 20 collateral
+        //perform partial take for 20 collateral
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: alignment, let's leave space in comment too

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 19]
```diff
@@ -603,17 +616,17 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
...
             {
                 from:            _lender,
                 borrower:        _borrower2,
                 maxCollateral:   20 * 1e18,
-                bondChange:      1.239648105429489010 * 1e18,
-                givenAmount:     123.964810542948901000 * 1e18,
-                collateralTaken: 20 * 1e18,
+                bondChange:      0.121516198312897248 * 1e18,
+                givenAmount:     12.151619831289724800 * 1e18,
+                collateralTaken: 20.0 * 1e18,
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: any reason `20 * 1e18` is written as `20.0 * 1e18`? if none I propose to leave it `20`

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 20]
```diff
@@ -622,28 +635,28 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
                 borrower:       _borrower2,
                 active:         true,
                 kicker:         _lender,
-                bondSize:       99.773590525221705467 * 1e18,
+                bondSize:       98.655458618105113705 * 1e18,
                 bondFactor:     0.01 * 1e18,
-                kickTime:       block.timestamp - 5 hours,
-                kickPriceIndex: 3696
+                kickTime:       block.timestamp - 10 hours,
+                kickMomp:       9.721295865031779605 * 1e18
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: alignment

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol [hunk 21]
```diff
@@ -664,28 +677,28 @@ contract ERC20PoolLiquidationsTest is ERC20HelperContract {
                 borrower:       _borrower2,
                 active:         true,
                 kicker:         _lender,
-                bondSize:       160.516347691266666957 * 1e18,
+                bondSize:       104.609752335437078857 * 1e18,
                 bondFactor:     0.01 * 1e18,
-                kickTime:       block.timestamp - 5 hours,
-                kickPriceIndex: 3696
+                kickTime:       8640000,
+                kickMomp:       9.721295865031779605 * 1e18
```
Comment (grandizzy) [inline src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: alignment

File: src/_test/ERC721Pool/ERC721PoolLiquidations.t.sol [hunk 22]
```diff
@@ -184,20 +184,20 @@ contract ERC721PoolLiquidationsTest is ERC721HelperContract {
                 bondSize:       0,
                 bondFactor:     0,
                 kickTime:       0,
-                kickPriceIndex: 0
+                kickMomp:       0
```
Comment (grandizzy) [inline src/_test/ERC721Pool/ERC721PoolLiquidations.t.sol]: alignment

File: src/base/Pool.sol [hunk 23]
```diff
@@ -362,25 +362,29 @@ abstract contract Pool is Clone, Multicall, IPool {
...
         uint256 lup = _lup(poolState.accruedDebt);
         if (_collateralization(borrower.debt, borrower.collateral, lup) >= Maths.WAD) revert BorrowerOk();
 
-        loans.kick(
-            borrowerAddress_,
-            borrower.debt,
-            borrower.inflatorSnapshot,
-            poolState.rate
-        );
+        uint256 kickPenalty;
```
Comment (grandizzy) [inline src/base/Pool.sol]: this can be one liner `uint256 kickPenalty = loans.kick` or even better `poolState.accruedDebt += loans.kick()`

File: src/base/Pool.sol [hunk 24]
```diff
@@ -614,7 +618,7 @@ abstract contract Pool is Clone, Multicall, IPool {
             poolState.accruedDebt
         );
         _updatePool(poolState, newLup);
-
+
```
Comment (grandizzy) [inline src/base/Pool.sol]: remove extra spaces

File: src/libraries/Auctions.sol [hunk 25]
```diff
@@ -20,7 +20,7 @@ library Auctions {
         uint256 bondSize;        // liquidation bond size
         uint256 bondFactor;      // bond factor used to start liquidation
         uint128 kickTime;        // timestamp when liquidation was started
-        uint128 kickPriceIndex;  // HPB price index at liquidation kick time
+        uint256 kickMomp;        // Momp when liquidation was started
```
Comment (grandizzy) [inline src/libraries/Auctions.sol]: align comments  (`kickPriceIndex` was longer)

File: src/libraries/Auctions.sol [hunk 26]
```diff
@@ -137,7 +135,7 @@ library Auctions {
         Liquidation storage liquidation = self.liquidations[borrower_];
         liquidation.kicker         = msg.sender;
         liquidation.kickTime       = uint128(block.timestamp);
-        liquidation.kickPriceIndex = uint128(hpbIndex_);
+        liquidation.kickMomp       = momp_;
```
Comment (grandizzy) [inline src/libraries/Auctions.sol]: alignment, remove extra spaces

File: src/libraries/Auctions.sol [hunk 27]
```diff
@@ -223,7 +222,7 @@ library Auctions {
                 self.kickers[liquidation.kicker].locked = 0;
             }
             else self.kickers[liquidation.kicker].locked -= bondChange_;
-        }
+        }
```
Comment (grandizzy) [inline src/libraries/Auctions.sol]: remove extra space

File: src/libraries/BucketMath.sol [hunk 28]
```diff
@@ -120,7 +120,7 @@ library BucketMath {
         elapsedHours -= Maths.min(elapsedHours, 1e18);  // price locked during cure period
 
         int256 timeAdjustment = PRBMathSD59x18.mul(-1 * 1e18, int256(elapsedHours));
```
Comment (grandizzy) [inline src/libraries/BucketMath.sol:122]: don't think we need `PRBMathSD59x18.mul` just to change the sign of `elapsedHours` ? https://consensys.github.io/smart-contract-best-practices/development-recommendations/general/negative-int/
isn't cheaper to `-(elapsedHours * 1e18)`?

at the same time `2^(-x) = 1/(2^x)` I think it worth checking if we can rewrite this as
`32 * Maths.wmul(referencePrice, Maths.wdiv(1e18, PRBMathUD60x18.exp2(timeAdjustment * 1e18));`
and compare gas costs

File: src/libraries/Loans.sol [hunk 29]
```diff
@@ -61,13 +61,14 @@ library Loans {
         uint256 debt_,
         uint256 inflator_,
         uint256 rate_
-    ) internal {
+    ) internal returns (uint256 kickPenalty_){
```
Comment (grandizzy) [inline src/libraries/Loans.sol]: nit: alignment, need a space to `{`
let's update `kick` function description too by including the `*  @return kickPenalty_  Description here.`

**Review by grandizzy — CHANGES_REQUESTED** [review]
Looking good overall! Just couple of nit alignments and extra spaces + doc to update and some comments re `auctionPrice` calculation

File: src/libraries/BucketMath.sol (hunk 28 above)
Comment (ith-harvey) [reply src/libraries/BucketMath.sol:122]: Hmm I attempted:
```
int256 timeAdjustment = -(int256(elapsedHours) * 1e18);
```
and 
```
price_ = 32 * Maths.wmul(referencePrice, Maths.wdiv(1e18, uint256(PRBMathSD59x18.exp2(timeAdjustment * 1e18))));
```
Neither of which worked out for me, each caused tests to fail... any other ideas?

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 17 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol:619]: The pricing function changed... the new starting price is higher thus the take after 5 hours was not below the neutral price of the loan and it wasn't giving the bond a reward.

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 1 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: fixed : )

File: src/libraries/BucketMath.sol (hunk 28 above)
Comment (grandizzy) [reply src/libraries/BucketMath.sol:122]: try `int256 timeAdjustment = -(elapsedHours * 1e18);`

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 2 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: fixed : )

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 3 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: fixed : )

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 4 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: fixed : )

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 5 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: fixed : )

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 6 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: fixed : )

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 7 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: fixed : )

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 8 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: fixed : )

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 9 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: fixed : )

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 10 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: fixed : )

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 11 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: fixed : )

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 12 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: fixed : )

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 13 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: fixed : )

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 14 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: fixed : )

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 15 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: fixed : )

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 16 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: fixed : )

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 18 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: fixed : )

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 19 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: fixed : )

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 20 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: fixed : )

File: src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol (hunk 21 above)
Comment (ith-harvey) [reply src/_test/ERC20Pool/ERC20PoolLiquidations.t.sol]: fixed : )

File: src/_test/ERC721Pool/ERC721PoolLiquidations.t.sol (hunk 22 above)
Comment (ith-harvey) [reply src/_test/ERC721Pool/ERC721PoolLiquidations.t.sol]: fixed : )

File: src/base/Pool.sol (hunk 23 above)
Comment (ith-harvey) [reply src/base/Pool.sol]: fixed : )

File: src/base/Pool.sol (hunk 24 above)
Comment (ith-harvey) [reply src/base/Pool.sol]: fixed : )

File: src/libraries/Auctions.sol (hunk 25 above)
Comment (ith-harvey) [reply src/libraries/Auctions.sol]: fixed : )

File: src/libraries/Auctions.sol (hunk 26 above)
Comment (ith-harvey) [reply src/libraries/Auctions.sol]: fixed : )

File: src/libraries/Auctions.sol (hunk 27 above)
Comment (ith-harvey) [reply src/libraries/Auctions.sol]: fixed : )

File: src/libraries/Loans.sol (hunk 29 above)
Comment (ith-harvey) [reply src/libraries/Loans.sol]: fixed : )

**Review by grandizzy — APPROVED** [review]
LGTM, Merge it!

File: src/libraries/BucketMath.sol (hunk 28 above)
Comment (ith-harvey) [reply src/libraries/BucketMath.sol:122]: Spoke with George, decided it was OK to leave this as is for now.
